cupy/_core/core.pyx

```diff
@@ -3072,8 +3079,10 @@ cpdef min_scalar_type(a):
 
     .. seealso:: :func:`numpy.min_scalar_type`
     """
-    if isinstance(a, ndarray):
-        return a.dtype
+    cdef _ndarray_base arr = _convert_from_cupy_like(a, error=False)
+    if arr is not None:
+        return arr.dtype
```

Comment:
The thing is that `_array_info_from_nested_sequence` actually _requires_ a sequence.
We could use `_compute_concat_info_impl` instead.  But it currently assumes that an object with `__cupy_get_ndarray__` also has a `.shape` and `.dtype` attribute.

So, I assumed that just using the simple conversion pattern is good enough here.